clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/RetrieveAndPrintExecutor.scala

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.clio.client.dispatch
+import org.broadinstitute.clio.client.commands._
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import io.circe.Json
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Executor for all commands that retrieve some data from the
+  * clio-server and then print it as JSON.
+  */
+class RetrieveAndPrintExecutor(command: ClioCommand) extends Executor {
+  override def execute(webClient: ClioWebClient, ioUtil: IoUtil)(
+    implicit ec: ExecutionContext,
+    bearerToken: OAuth2BearerToken
+  ): Future[HttpResponse] = {
+    val responseFut = command match {
```

Comment:
This seems much better but is there a way to put the `match` inside `ClioCommand`?

Reply:
We could add a method to the `ClioCommand` trait, but it's more idiomatic to separate the data definitions and the operations over them. `ClioCommand` also has to be sealed for caseapp to work, so I'm trying to keep as much code out of that file as possible to minimize its bloat as we add more commands.

Reply:
ok